command_line/anvil_correction.py

```diff
@@ -0,0 +1,318 @@
+    print(flex.max(correction), flex.min(correction))
+    # Remove redundant things that scale with N.
+    del l0, l1
+
+    # Correct the measured intensities for this attenuation.
+    for col in "intensity.prf.value", "intensity.sum.value":
+        try:
+            reflections[col].set_selected(sel, refls_sel[col] * correction)
+        except KeyError:
+            pass
+    # Correct the measured variances accordingly.
+    variance_correction = flex.pow2(correction)
+    del correction
+    for col in "intensity.prf.variance", "intensity.sum.variance":
+        try:
+            reflections[col].set_selected(sel, refls_sel[col] * variance_correction)
+        except KeyError:
+            pass
+
+
+def run(args=None, phil=phil_scope):  # type: (List[str], libtbx.phil.scope) -> None
+    """
+    Run dials.anvil_correction as from the command line.
+
+    Take integrated experiment lists and reflection tables and correct the all the
+    integrated intensities for the estimated attenuation by the diamond anvils.
+
+    Args:
+        args: The arguments supplied by the user (default: sys.argv[1:]).
+        phil: The PHIL scope definition (default: phil_scope, the master PHIL scope
+              for this program).
+    """
+    usage = "dials.%s [options] integrated.expt integrated.refl" % filename_root
+
+    parser = OptionParser(
+        usage=usage,
+        phil=phil,
+        read_reflections=True,
+        read_experiments=True,
+        check_format=False,
+        epilog=__doc__,
+    )
+
+    params, options = parser.parse_args(args=args, show_diff_phil=False)
+
+    # Configure the logging.
+    dials.util.log.config(options.verbose, logfile=params.output.log)
+
+    # Log the difference between the PHIL scope definition and the active PHIL scope,
+    # which will include the parsed user inputs.
+    diff_phil = parser.diff_phil.as_str()
+    if diff_phil:
+        logger.info("The following parameters have been modified:\n%s", diff_phil)
+
+    # Check that at least one reflection table and experiment list have been provided.
+    input_errors = []
+    if not params.input.experiments:
+        input_errors.append(
+            "Please provide at least one valid experiment list (.expt) file."
+        )
+    if not params.input.reflections:
+        input_errors.append(
+            "Please provide at least one valid reflection table (.refl) file."
+        )
+    if input_errors:
+        sys.exit("\n".join([parser.format_help()] + input_errors))
+
+    # These functions are commonly used to collate the input.
+    experiments = flatten_experiments(params.input.experiments)
+    reflections_list = flatten_reflections(params.input.reflections)
+    reflections_list = parse_multiple_datasets(reflections_list)
+    reflections_list = sort_tables_to_experiments_order(reflections_list, experiments)
+    # Work around parse_multiple_datasets dropping unindexed reflections.
```

Comment:
I think this workaround should be before the call to parse_multiple_datasets?

Reply:
Oh dear, thanks, that was a slip.  Well done for spotting that.